src/constraints.jl

```diff
@@ -0,0 +1,86 @@
+export
+    apply_constraints!,
+    SHAKE
+
+
+"""
+    apply_constraints!(system, old_coords, dt)
+
+Applies all the bond and angle constraints associated with the system.
+"""
+
+function apply_constraints!(sys, old_values, dt)
+    for constraint in sys.constraints
+        apply_constraints!(sys, constraint, old_values, dt)
+    end
+end
+
+"""
+    SHAKE(d, bond_list)
+
+Constrains a set of bonds to a particular distance d.
+"""
+struct SHAKE{D, B}
+    d::D
```

Comment:
This could be called `dists`.

It would also be good to have a constructor from keyword arguments like https://github.com/JuliaMolSim/Molly.jl/blob/master/src/interactions/harmonic_bond.jl#L17.

The docstring also needs updating.